### ISSUE TITLE
Add `none` nictype to LXD

### ISSUE THREAD

**Issue by alejandro-perez** [lxc/lxd#4046]
Hi, I have a use case where I need to include the container into the hosts' network, but the host is connected though a Wifi device. Since bridged and maclan would not work with Wifi devices, in LXC I used to use `ǹone` as the network type:
```
              none: will cause the container to share the host's network
              namespace. This means the host network devices are usable in
              the container. It also means that if both the container and
              host have upstart as init, 'halt' in a container (for
              instance) will shut down the host.
```
This made my container to have all the interfaces the host has, and hence, could use the WLAN0 directly. 

But it either seems not possible from LXD (I even tried with lxc.raw) or I could not find a way.

**Comment by stgraber** [comment]
LXD doesn't allow you to pick and choose what namespaces to use the way low level LXC does.

We don't have any plans to support sharing the network namespace with the host as this tends to be very dangerous and also doesn't fit in the general configuration model of LXD.
Sharing netns with the host bridges far more than just your network interfaces, it also makes your container share all abstract unix sockets as well as kernel uevents, which tends to confuse processes on host and container.

Most people in your situation will just use a routed subnet assigned to a LXD managed bridge or will have the host do port forwarding to the container for those port you care about.

If you absolutely must do it this way, it can still be achieved through raw.lxc but then you're in entirely unsupported territory. If you must do this, then create a file which contains:

    lxc.network.0.type=none

Or if on liblxc 2.1 or higher:

    lxc.net.0.type=none

And then set raw.lxc to "lxc.include=/path/to/that/file". This should override your liblxc configuration and force the key to be set to none.

**Comment by alejandro-perez** [comment]
Thanks @stgraber for the explanation. I see now why host network is not something you want, at least not in an easy way.

I tried with `raw.lxc: lxc.net.0.type=none` directly, but that does not work. I didn't think of using a include, that makes sense.

In any case considering what you just said, I'll try to avoid the use of `none`. What I can tell is that it worked fairly well with plain LXC, I had no problems. I've seen the problems you mentioned if I try to use plain chroot with some binds thought, so I can imagine what you said.

Port forwarding is another option, specially if LXD had a way of doing it automatically. I made use of https://github.com/patrakov/lxc-expose-port/blob/master/lxc-expose-port when using LXC, is there something similar to `lxc.net.0.script.up` in LXD?
